app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt

```diff
@@ -207,15 +364,25 @@ private fun ProfileHeader(
 
                         onUpdateUsername(userNameEditing)
                         context.showToast(message = "Your username has been updated!")
+                    } else if (userNameEditing.length > 30) {
+                        userNameEditing = userNameDisplay
+
+                        context.showToast(message = "Username cannot be longer than 30 characters")
                     } else {
+                        userNameEditing = userNameDisplay
+
                         context.showToast(message = "Username cannot be empty!")
```

Comment:
Could we make these messages more... passive? Without any exclamation marks: `?!.` and only with `,` if required.
Everywhere within this file.